### PR TITLE
Correctly handle `--help` flags when `argv.remainder!` is used

### PR DESCRIPTION
**Pull request by amorde**
Supersedes https://github.com/CocoaPods/CocoaPods/pull/8571

### CODE REVIEW

**Comment by amorde** [comment]
Tests passed except for old Ruby versions that aren't on Travis anymore, I can update the config but maybe we should drop support for those versions?

**Comment by dnkoutso** [comment]
Yes lets do the same set CocoaPods/CocoaPods does?

**Comment by amorde** [comment]
Yup I'll do that in another PR before merging this

**Comment by dnkoutso** [comment]
An attempt was made here https://github.com/CocoaPods/CLAide/pull/86

**Comment by dnkoutso** [comment]
@amorde add a `CHANGELOG` bug fix entry and then point to `1-0-stable` instead as I will do a 1.0.3 release for this.

**Comment by amorde** [comment]
@dnkoutso `1-0-stable` branch is out of date, right now it's pointing to 1.0.1 instead of 1.0.2

Should I update it?

**Comment by amorde** [comment]
Updated the stable branch, and re-targeted this PR

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,7 +8,15 @@
 
 ##### Bug Fixes
 
-* None.  
+* Correctly handle `--help` flags when using `argv.remainder!` after initialization  
+  [Eric Amorde](https://github.com/amorde),
+  [tripleCC](https://github.com/tripleCC)
+  [#87](https://github.com/CocoaPods/CLAide/pull/87)
+
+
```
Comment (dnkoutso) [inline CHANGELOG.md]: nit empty line

Comment (amorde) [reply CHANGELOG.md]: are you saying there's an extra line? I copied the style from the other sections

**Pull request merged**